src/meta-srv/src/region/supervisor.rs

```diff
@@ -389,75 +422,121 @@ impl RegionSupervisor {
             .context(error::MaintenanceModeManagerSnafu)
     }
 
-    async fn do_failover(&mut self, datanode_id: DatanodeId, region_id: RegionId) -> Result<()> {
-        let count = *self
-            .failover_counts
-            .entry((datanode_id, region_id))
-            .and_modify(|count| *count += 1)
-            .or_insert(1);
+    async fn select_peers(
+        &self,
+        from_peer_id: DatanodeId,
+        regions: &[RegionId],
+    ) -> Result<Vec<(RegionId, Peer)>> {
+        match &self.selector {
+            RegionSupervisorSelector::NaiveSelector(selector) => {
+                let opt = SelectorOptions {
+                    min_required_items: regions.len(),
+                    allow_duplication: true,
+                    exclude_peer_ids: HashSet::from([from_peer_id]),
+                };
+                let peers = selector.select(&self.selector_context, opt).await?;
+                ensure!(
+                    peers.len() == regions.len(),
+                    error::PeerCountMismatchSnafu {
+                        expected: regions.len(),
+                        actual: peers.len(),
+                    }
+                );
+                let region_peers = regions
+                    .iter()
+                    .zip(peers)
+                    .map(|(region_id, peer)| (*region_id, peer))
+                    .collect::<Vec<_>>();
+
+                Ok(region_peers)
+            }
+            RegionSupervisorSelector::RegionStatAwareSelector(selector) => {
+                let peers = selector
+                    .select(&self.selector_context, from_peer_id, regions)
+                    .await?;
+
+                ensure!(
+                    peers.len() == regions.len(),
+                    error::PeerCountMismatchSnafu {
+                        expected: regions.len(),
+                        actual: peers.len(),
+                    }
+                );
```

Comment:
```suggestion
                );

```

Reply:
Resolved in [5cc7d5d](https://github.com/GreptimeTeam/greptimedb/pull/5990/commits/5cc7d5d18ad1d69199e2a6d2d67a347b11373529)